gravitee-apim-plugin/gravitee-apim-plugin-entrypoint/gravitee-apim-plugin-entrypoint-webhook/src/main/java/io/gravitee/plugin/entrypoint/webhook/WebhookEntrypointConnector.java

```diff
@@ -97,14 +103,51 @@ public Completable handleResponse(final ExecutionContext ctx) {
                         ctx
                             .response()
                             .messages()
-                            .flatMapCompletable(message -> sendAndDiscard(requestUri, httpClient, message))
+                            .compose(applyStopHook())
+                            .flatMapCompletable(
+                                message -> {
+                                    if (message.error()) {
+                                        return Completable.error(new Exception(message.content().toString()));
+                                    }
+                                    return sendAndDiscard(requestUri, httpClient, message);
+                                }
+                            )
                             .doFinally(httpClient::close)
                             .toFlowable()
                     );
             }
         );
     }
 
+    @Override
+    public WebhookEntrypointConnector preStop() throws Exception {
+        // Note: for now, we just drop an error to stop message consumption and let the error propagate to the dispatcher. This is likely to change when subscription lifecycle will be properly handled.
+        stopHook.onError(new Exception(STOPPING_MESSAGE));
```

Comment:
Instead of executing specific onError, you could do the samething as others and sending back a onError message which will be handle properly by the code line 110.